### ISSUE TITLE
DataList - invalid/incorrect aria attributes in examples

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly#7274]
**Describe the enhancement or change**
- Per [naming role guidance from W3C](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#naming_role_guidance), listitems should not have aria labeling even if technically it may work. We should remove any aria-label, aria-labelledby, and/or aria-describedby from any `li` items in this component.
- In expandable examples, the toggle has an invalid Id passed to the aria-labelledby attribute - the ID being reference is `...item[1|2|3|...]`, but there should be a hyphen between the word "item" and the number e.g. "data-list-expandable-compact-item-1".

**Any other information?**
